rust/kernel/init/macros.rs

```diff
@@ -99,6 +99,158 @@ macro_rules! __pin_data {
             @pinned_drop($($pinned_drop)?),
         );
     };
+    (find_pinned_fields:
+        @struct_attrs($($struct_attrs:tt)*),
+        @vis($vis:vis),
+        @name($name:ident),
+        @impl_generics($($impl_generics:tt)*),
+        @ty_generics($($ty_generics:tt)*),
+        @where($($whr:tt)*),
+        // We found a PhantomPinned field, this should generally be pinned!
+        @fields_munch($field:ident : ::core::marker::PhantomPinned, $($rest:tt)*),
```

Comment:
Maybe instead of duplicating these checks, maybe just a single check for `$($(::)? core::)? $(marker::)? PhantomPinned`?

No body should be defining a custom type named `PhantomPinned` anyway.

Reply:
they *should* not, but it might result from some macro that takes some name from somewhere etc. (yes this is pretty contrived, but I still want to catch the simplest cases). Ideally we could use some type system solution to catch exactly the right cases, but I could not come up with one.